custom_components/airthings_wave/airthings.py

```diff
@@ -192,152 +206,158 @@ def __init__(self, scan_interval, mac=None):
         self.scan_interval = scan_interval
         self.last_scan = -1
         self._dev = None
-
-    def _parse_serial_number(self, manufacturer_data):
-        try:
-            (ID, SN, _) = struct.unpack("<HLH", manufacturer_data)
-        except Exception as e:  # Return None for non-Airthings devices
-            return None
-        else:  # Executes only if try-block succeeds
-            if ID == 0x0334:
-                return SN
-
-    def find_devices(self, scans=50, timeout=0.1):
+        self._command_data = None
+
+    def notification_handler(self, sender, data):
+        _LOGGER.debug("Notification handler: {0}: {1}".format(sender, data))
+        self._command_data = data
+        self._event.set()
+    
+    async def find_devices(self, scans=2, timeout=5):
         # Search for devices, scan for BLE devices scans times for timeout seconds
         # Get manufacturer data and try to match it to airthings ID.
-        scanner = btle.Scanner()
+        
+        _LOGGER.debug("Scanning for airthings devices")
         for _count in range(scans):
-            advertisements = scanner.scan(timeout)
+            advertisements = await BleakScanner.discover(timeout)
             for adv in advertisements:
-                sn = self._parse_serial_number(adv.getValue(btle.ScanEntry.MANUFACTURER))
-                if sn is not None:
-                    if adv.addr not in self.airthing_devices:
-                        self.airthing_devices.append(adv.addr)
+                if 820 in adv.metadata["manufacturer_data"]: # TODO: Not sure if this is the best way to identify Airthings devices
```

Comment:
Any chance this TODO can be resolved?

Reply:
@lymanepp That was a note to myself, but it can probably be removed now that it has worked for others. As far as I can tell I am using the same method as before, it is just that bleak returns the values quite differently than bluepy.